### DISCUSSION TITLE
How to use Advance Mass Scaling (/AMS) with k-file

### DISCUSSION THREAD

**Discussion by Hyp5891** [OpenRadioss/OpenRadioss#1976] [Q&A]
Hi,

I would like to try /AMS over /CST mass-scaling option to study wether this further reduces run-time.
I build my model using ls-prepost means I have a k-file as input.

As far as i understood from the documentation one has to place /AMS keyword in the starter (0000.rad) and /DT/AMS into the engine-file ( 0001.rad).

My problem now is when coming from a k.-file there is no 0000.rad which i could modify before running.
Is there a way to somehow write a special section into the k.-File that the translater (.k --> .rad) will recognize as radioss input or some other solution for this?

Thanks in advance!

Best regards,
Christian

**Comment by PaulAltair** [comment]
You can use the *INCLUDE_RADIOSS functionality to do this, *INCLUDE_RADIOSS is a 'dyna' card for OpenRadioss that allows you to include Radioss format input in your otherwise DYNA format deck by adding it in an include file (create an include file with your /AMS keyword in Radioss format and include it via the *INCLUDE_RADIOSS card in the DYNA deck) I have a attached a very simple example here modifying the dyna format tensile test from the examples
[Version_With_AMS_Using_INCLUDERADIOSS.zip](https://github.com/OpenRadioss/OpenRadioss/files/12910896/Version_With_AMS_Using_INCLUDERADIOSS.zip)

In order to use this, you will need to run the starter on the key/k file (that is calling the include) and before running the engine, substitute in the engine file including the /DT/AMS card (as in the zip file)

**Comment by MarkusSchulten** [comment]
Hi,
is it possible to use Sol2sph with *INCLUDE_RADIOSS to modify a k-file with solid elements?
Kindly regards
Markus

**Reply by elequiniou** [reply]
Hi Markus, please, did you give a try since then ? 
I don't see why it should not work, but I am not sure this combination has been extensively tested yet. Please, let us know if you encounter any issue.
Thanks a lot,
Eric
